### ISSUE TITLE
scrollToOffset/index in horizontal direction with paging is wrong

### ISSUE THREAD

**Issue by segicm** [Shopify/flash-list#613]
## Current behavior

I am having a long list of items rendered in a Horizontal direction. On a button press, I want to scroll to a certain item in the list. The problem is that the "scrollToOffset" and "scrollToIndex" are not scrolling me to the middle of the item as it is shown on the screenshot. The first few items are centered but if I try to scroll to a 15. item, for example, then it gets it wrong.

![Simulator Screen Shot - iPhone 12 Pro - 2022-09-25 at 11 12 36](https://user-images.githubusercontent.com/30130573/192136471-fd31d5cd-b1ba-408b-a243-d36e90875cdf.png)

My code:

```
<FlashList
    ref={flatListRef}
    data={data}
    renderItem={renderItem}
    horizontal
    pagingEnabled
    showsHorizontalScrollIndicator={false}
    estimatedItemSize={400}
    viewabilityConfigCallbackPairs={viewabilityConfigCallbackPairs.current}
  />
```
The render item wrapper has a width equal to the width of the device screen.

This is how I am trying to scroll to the item.
```
flatListRef?.current?.scrollToOffset({
    offset: width * index,
    animated: true,
  });
```

## Expected behavior

This is how it looks when I scroll to the first five items, and this is how it should be for the all items on the list.
![Simulator Screen Shot - iPhone 12 Pro - 2022-09-25 at 11 12 45](https://user-images.githubusercontent.com/30130573/192136550-4d9ea626-9a41-4630-8d19-c8ec6c8f66ca.png)

## Platform:

- [x] iOS
- [ ] Android

## Environment

1.1.0

**Comment by naqvitalha** [comment]
Can you create an expo snack for this? I can take a look

**Comment by segicm** [comment]
I have tried to recreate the problem, that I have, on the Expo Snack and everything worked fine. So, I started digging more into the problem on my side and discovered that my custom hook created this problem. I think the problem is that the hook is rerendering the item at the same time when the scrollToOffset is working.

Here is an example on the snack: https://snack.expo.dev/@sega93/upbeat-strawberries

(Run it on iOS or Android mode)
